controllers/gitopsservice_controller.go

```diff
@@ -321,6 +321,41 @@ func (r *ReconcileGitopsService) reconcileDefaultArgoCDInstance(instance *pipeli
 	if err := controllerutil.SetControllerReference(instance, defaultArgoCDInstance, r.Scheme); err != nil {
 		return reconcile.Result{}, err
 	}
+	//to add infra nodeselector to default argocd pods
+	if instance.Spec.InfraNodeEnabled {
+		defaultArgoCDInstance.Spec.NodePlacement = &argoapp.ArgoCDNodePlacementSpec{
+			NodeSelector: map[string]string{
+				"node-role.kubernetes.io/infra": "",
+			},
+		}
+	}
+	if len(instance.Spec.Tolerations) > 0 {
+		if defaultArgoCDInstance.Spec.NodePlacement == nil {
+			defaultArgoCDInstance.Spec.NodePlacement = &argoapp.ArgoCDNodePlacementSpec{
+				Tolerations: instance.Spec.Tolerations,
+			}
+		} else {
+			defaultArgoCDInstance.Spec.NodePlacement.Tolerations = instance.Spec.Tolerations
+		}
+	}
+
+	//to add infra nodeselector to default argocd pods
+	if instance.Spec.InfraNodeEnabled {
+		defaultArgoCDInstance.Spec.NodePlacement = &argoapp.ArgoCDNodePlacementSpec{
+			NodeSelector: map[string]string{
+				"node-role.kubernetes.io/infra": "",
+			},
+		}
+	}
+	if len(instance.Spec.Tolerations) > 0 {
+		if defaultArgoCDInstance.Spec.NodePlacement == nil {
+			defaultArgoCDInstance.Spec.NodePlacement = &argoapp.ArgoCDNodePlacementSpec{
+				Tolerations: instance.Spec.Tolerations,
+			}
+		} else {
+			defaultArgoCDInstance.Spec.NodePlacement.Tolerations = instance.Spec.Tolerations
+		}
+	}
```

Comment:
@saumeya Looks like this block of code may have been repeated accidentally

Reply:
thanks a lot for noticing jaideep, must have happened with some rebase. i'll correct it